hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -659,6 +659,10 @@ private void processDataBlock(HoodieDataBlock dataBlock, Option<KeySpec> keySpec
    */
   protected abstract void processNextDeletedRecord(DeleteRecord deleteRecord);
 
+  public Set<String> getDeletedRecordKeys() {
+    throw new HoodieException("Inherited class needs to override to provide a concrete implementation");
+  }
```

Comment:
I would say we add a specific delete records callback just in `HoodieUnMergedLogRecordScanner` to collect the deleted keys, so that we can avoid the changes for this base class and also the member variable in `HoodieUnMergedLogRecordScanner`.